README.md

```diff
@@ -29,46 +30,69 @@ Installation of the [npm package](https://npmjs.org/package/reaclette):
 ```js
 import React from "react";
 import { render } from "react-dom";
-import { injectState, provideState } from "reaclette";
-
-const wrapComponentWithState = provideState({
-  initialState: () => ({ counter: 0 }),
-  effects: {
-    addOne: () => (state, props) => ({ counter: state.counter + 1 }),
-  },
-  computed: {
-    square: (state, props) => state.counter * state.counter,
+import { withStore } from "reaclette";
+
+const Component = withStore(
+  {
+    initialState: props => ({ counter: 0 }),
+    effects: {
+      addOne() {
+        this.state.counter += 1;
+      },
+    },
+    computed: {
+      square: (state, props) => state.counter * state.counter,
+    },
   },
-});
-
-const Parent = wrapComponentWithState(() => <Child />);
-
-const Child = injectState(({ effects, state }) => (
-  <div>
-    <p>Our counter is at: {state.counter}</p>
-    <p>Its squared value is: {state.square}</p>
-    <p>
-      <button onClick={effects.addOne}>Add one</button>
-    </p>
-  </div>
-));
+  (store, props) => (
+    <div>
+      <p>Our counter is at: {store.state.counter}</p>
+      <p>Its squared value is: {store.state.square}</p>
+      <p>
+        <button onClick={store.effects.addOne}>Add one</button>
+      </p>
+    </div>
+  )
+);
 
-render(<Parent />, document.body);
+render(<Component />, document.body);
 ```
 
 ## API
 
 ```js
-import { provideState, injectState } from "reaclette";
+import { withStore } from "reaclette";
 ```
 
-### `provideState(options) => (Component => Component)`
+### `withStore(options, renderFunction)`
 
 Create a decorator that associates a React component with a store.
 
-`options` is an object which can contain the following properties.
+`renderFunction`
+
+A function that receives `store` and `props` in parameters and returns a `React node` like so:
+
+```js
+(store, props) => ReactNode;
+```
+
+- `store`
+  Is an object containing the following properties:
+
+  - [`state`](#initial-state)
```

Comment:
This link is incorrect.

Reply:
But I don't know what it should link to.

Maybe we should write something like:

```md
- `state` (initialized by [`initialState`](#initial-state), mutated by `effects` and augmented by `computed`)
```